File: CoreTransformer/TransitionSizeDistribution.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

''' <summary>
''' Transition Size Distribution class
''' </summary>
''' <remarks></remarks>
Friend Class TransitionSizeDistribution

    Private m_TransitionSizeDistributionId As Integer
    Private m_StratumId As Nullable(Of Integer)
    Private m_Iteration As Nullable(Of Integer)
    Private m_Timestep As Nullable(Of Integer)
    Private m_TransitionGroupId As Integer
    Private m_MinimumSize As Double
    Private m_MaximumSize As Double
    Private m_RelativeAmount As Double
    Private m_Proportion As Double

    Public Sub New(
        ByVal transitionSizeDistributionId As Integer,
        ByVal stratumId As Nullable(Of Integer),
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal transitionGroupId As Integer,
        ByVal maximumSize As Double,
        ByVal relativeAmount As Double)

        Me.m_TransitionSizeDistributionId = transitionSizeDistributionId
        Me.m_StratumId = stratumId
        Me.m_Iteration = iteration
        Me.m_Timestep = timestep
        Me.m_TransitionGroupId = transitionGroupId
        Me.m_MaximumSize = maximumSize
        Me.m_RelativeAmount = relativeAmount

    End Sub

    ''' <summary>
    ''' Gets the transition size distribution Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property TransitionSizeDistributionId As Integer
        Get
            Return Me.m_TransitionSizeDistributionId
        End Get
    End Property
    ''' <summary>
    ''' Gets the stratum Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property StratumId As Nullable(Of Integer)
        Get
            Return Me.m_StratumId
        End Get
    End Property

    ''' <summary>
    ''' Gets the iteration
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property Iteration As Nullable(Of Integer)
        Get
            Return Me.m_Iteration
        End Get
    End Property

    ''' <summary>
    ''' Gets the timestep
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property Timestep As Nullable(Of Integer)
        Get
            Return Me.m_Timestep
        End Get
    End Property

    ''' <summary>
    ''' Gets the transition group Id
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

    ''' <summary>
    ''' Gets or sets the minimum size
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property MinimumSize As Double
        Get
            Return Me.m_MinimumSize
        End Get
        Set(value As Double)
            Me.m_MinimumSize = value
        End Set
    End Property

    ''' <summary>
    ''' Gets the maximum size
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property MaximumSize As Double
        Get
            Return Me.m_MaximumSize
        End Get
    End Property

    ''' <summary>
    ''' Gets the relative amount
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public ReadOnly Property RelativeAmount As Double
        Get
            Return Me.m_RelativeAmount
        End Get
    End Property

    ''' <summary>
    ''' Gets or sets the proportion
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Proportion As Double
        Get
            Return Me.m_Proportion
        End Get
        Set(value As Double)
            Me.m_Proportion = value
        End Set
    End Property

End Class

File: Runtime/TransitionSizeDistribution.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class TransitionSizeDistribution
    {
        private int m_TransitionSizeDistributionId;
        private int? m_StratumId;
        private int? m_Iteration;
        private int? m_Timestep;
        private int m_TransitionGroupId;
        private double m_MinimumSize;
        private double m_MaximumSize;
        private double m_RelativeAmount;
        private double m_Proportion;

        public TransitionSizeDistribution(
            int transitionSizeDistributionId, int? stratumId, int? iteration, int? timestep, 
            int transitionGroupId, double maximumSize, double relativeAmount)
        {
            this.m_TransitionSizeDistributionId = transitionSizeDistributionId;
            this.m_StratumId = stratumId;
            this.m_Iteration = iteration;
            this.m_Timestep = timestep;
            this.m_TransitionGroupId = transitionGroupId;
            this.m_MaximumSize = maximumSize;
            this.m_RelativeAmount = relativeAmount;
        }

        public int TransitionSizeDistributionId
        {
            get
            {
                return this.m_TransitionSizeDistributionId;
            }
        }

        public int? StratumId
        {
            get
            {
                return this.m_StratumId;
            }
        }

        public int? Iteration
        {
            get
            {
                return this.m_Iteration;
            }
        }

        public int? Timestep
        {
            get
            {
                return this.m_Timestep;
            }
        }

        public int TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }

        public double MinimumSize
        {
            get
            {
                return this.m_MinimumSize;
            }
            set
            {
                this.m_MinimumSize = value;
            }
        }

        public double MaximumSize
        {
            get
            {
                return this.m_MaximumSize;
            }
        }

        public double RelativeAmount
        {
            get
            {
                return this.m_RelativeAmount;
            }
        }

        public double Proportion
        {
            get
            {
                return this.m_Proportion;
            }
            set
            {
                this.m_Proportion = value;
            }
        }
    }
}
